packages/blocks/jira-cloud/src/lib/common/index.ts

```diff
@@ -74,13 +74,32 @@ export async function searchUserByCriteria(
 }
 
 export async function getProjects(auth: JiraAuth): Promise<JiraProject[]> {
-  const response = await sendJiraRequest({
-    url: 'project/search',
-    method: HttpMethod.GET,
-    auth: auth,
-  });
+  let startAt = 0;
+  const results: JiraProject[] = [];
+  let total = undefined as number | undefined;
+  const pageSize = 50;
+
+  do {
```

Comment:
the pagination looks slightly different depending on which endpoints youre looking at, so ive not made a generic function just yet, we can see with other functions what the similarities are and what not.